### DISCUSSION TITLE
Ability to create my own test suite

### DISCUSSION THREAD

**Discussion by jriordan22849** [forcedotcom/salesforcedx-vscode#5779] [Ideas]
**Is your feature request related to a problem? Please describe.**
Yes, more flexibility in configuring unit tests. I would like to configure tests that test the broad range of a new feature or a bug fix. I do not want to execute a whole test class but partial tests in that test class. I would like to have the ability to configure which tests to run from multiple tests classes.
 
**Describe the solution you'd like**
Instead of running a single test or a test class, I would to run my own test suite that consists of multiple unit tests across multiple test classes. The suite should be configurable via a file which I can update. 
**Describe alternatives you've considered**
Currently, we can only run a single test or a test class. 
**Additional context**

**Comment by jefersonchaves** [comment]
Hi, is there any update around this? As far as I understand this is available on sfdx but now requires to be surfaced in VsCode.

**Comment by lcampos** [comment]
Hi @jriordan22849 , thanks for logging this. Just to clarify, does this mean you want to use the existing Apex Test Suite (https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_apextestsuite.htm) or something different than that ?

**Comment by jriordan22849** [comment]
Hi @lcampos , ApexTestSuite is only been able to configure a set of apex classes. I would like to configure what tests classes to run and/or test methods in a class. In some scenarios, I would only like to run certain classes from multiple classes instead of all the test methods in the listed classes. Having the ability to configure which class and/or methods would be very beneficial. 

Thanks

**Comment by jriordan22849** [comment]
hey @lcampos , is there any update in relation to this?

**Comment by smaddox-sf** [comment]
Hi @jriordan22849 - Thank you for the feature request.  This isn't on our [near-term roadmap](https://github.com/forcedotcom/salesforcedx-vscode/wiki/Roadmap), but we'll keep the feedback in mind for the future.

**Comment by mshanemc** [comment]
you can make your own test suite, but the members must be classes (so all the tests in a class), not individual tests within classes.

That's just how the API works (see
https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_apextestsuite.htm)

To do test-level suites, you'd need a change beyond what the extensions can do as a consumers of the API
